src/engine/renderer/tr_shade.cpp

```diff
@@ -36,29 +36,33 @@ This file deals with applying shaders to surface data in the tess struct.
 static void EnableAvailableFeatures()
 {
 	glConfig2.dynamicLight = r_dynamicLight.Get();
-	glConfig2.staticLight = r_staticLight.Get();
 
-	if ( glConfig2.dynamicLight || glConfig2.staticLight )
+	if ( glConfig2.dynamicLight )
 	{
 		if ( r_dynamicLightRenderer.Get() == Util::ordinal( dynamicLightRenderer_t::TILED ) )
 		{
 			if ( !glConfig2.textureFloatAvailable )
 			{
 				Log::Warn( "Tiled dynamic light renderer disabled because GL_ARB_texture_float is not available.");
 				glConfig2.dynamicLight = false;
-				glConfig2.staticLight = false;
 			}
 
 			// See below about ALU instructions on ATI R300 and Intel GMA 3.
 			if ( !glConfig2.glCoreProfile && glConfig2.maxAluInstructions < 128 )
 			{
 				Log::Warn( "Tiled dynamic light rendered disabled because GL_MAX_PROGRAM_ALU_INSTRUCTIONS_ARB is too small: %d", glConfig2.maxAluInstructions );
 				glConfig2.dynamicLight = false;
-				glConfig2.staticLight = false;
 			}
 		}
 	}
 
+	// Enable different kinds of dynamic lights.
+	glConfig2.animatedLight = glConfig2.dynamicLight && r_animatedLight.Get();
+	glConfig2.staticLight = glConfig2.dynamicLight && r_staticLight.Get();
+
+	// Finally tell the engine the dynamic light code is needed, whatever the kind of light.
+	glConfig2.dynamicLight = glConfig2.animatedLight || glConfig2.staticLight;
```

Comment:
I would prefer taking this value directly from r_dynamicLights without checking other cvars. Running the dynamic light code path with 0 lights is a valid debugging use case. Also this would let the debugging cvar for dropping dynamic lights be non-latch which really improves the velocity of debugging.